File: scala/scala-impl/test/org/jetbrains/plugins/scala/lang/actions/editor/enter/multiline_string/MultiLineStringAlignInfixAllTest.java
package org.jetbrains.plugins.scala.lang.actions.editor.enter.multiline_string;

import com.intellij.psi.codeStyle.CommonCodeStyleSettings;
import junit.framework.Test;
import org.jetbrains.plugins.scala.lang.actions.editor.enter.AbstractEnterActionTestBase;
import org.jetbrains.plugins.scala.lang.formatting.settings.ScalaCodeStyleSettings;
import org.junit.runner.RunWith;
import org.junit.runners.AllTests;

@RunWith(AllTests.class)
public class MultiLineStringAlignInfixAllTest extends AbstractEnterActionTestBase {
  public static final String DATA_PATH = "/actions/editor/enter/multiLineStringData/alignInfix/indentAndMargin";

  public MultiLineStringAlignInfixAllTest() {
    super(DATA_PATH);
  }

  @Override
  protected void setSettings() {
    super.setSettings();
    final CommonCodeStyleSettings settings = getCommonSettings();
    final ScalaCodeStyleSettings scalaSettings = settings.getRootSettings().getCustomSettings(ScalaCodeStyleSettings.class);

    scalaSettings.MULTILINE_STRING_SUPPORT = ScalaCodeStyleSettings.MULTILINE_STRING_ALL;
    scalaSettings.MULTI_LINE_STRING_MARGIN_INDENT = 3;
    settings.ALIGN_MULTILINE_BINARY_OPERATION = true;
  }

  public static Test suite() {
    return new MultiLineStringAlignInfixAllTest();
  }
}

File: scala/scala-impl/test/org/jetbrains/plugins/scala/lang/actions/editor/enter/multiline_string/MultiLineStringAlignInfixAllTest.scala
package org.jetbrains.plugins.scala.lang.actions.editor.enter.multiline_string

import java.nio.file.Path

class MultiLineStringAlignInfixAllTest extends MultiLineStringEnterHandlerTestBase {
  override protected def relativeTestDataPath: Path = Path.of("actions", "editor", "enter", "multiLineStringData", "alignInfix", "indentAndMargin")

  override protected def setUp(): Unit = {
    super.setUp()
    scalaCodeStyleSettings.MULTILINE_STRING_CLOSING_QUOTES_ON_NEW_LINE = true
    scalaCodeStyleSettings.MULTILINE_STRING_INSERT_MARGIN_ON_ENTER = true
    scalaCodeStyleSettings.MULTILINE_STRING_MARGIN_INDENT = 3
    commonCodeStyleSettings.ALIGN_MULTILINE_BINARY_OPERATION = true
  }
}
